middleware/common/source/communication/tcp.cpp

```diff
@@ -85,11 +85,14 @@ namespace casual
                   canonical = AI_CANONNAME,
                };
 
+               //! incicate that the enum is used as a flag (g++11 force to deinfe it)
+               [[maybe_unused]] consteval void casual_enum_as_flag( Flag) {};
+
```

Comment:
Just out of curiosity, why did we need to _define_ it here and not on the other places this construction is used?

Reply:
Because of a bug in g++11 :) Which I think is triggered because the compiler can see that it is not used (unnamed ns). It is used compile-time though which `consteval` strongly indicate. This is why we need the `[[maybe_unused]]` attribute, which we should not need. Then g++11 say it is not _defined_, so I defined it... 
In short, I think this bug "appear" because of the _unnamed ns context_.

Reply:
g++-13 does not need the [[maybe_unused]].
It gives the warning 
middleware/common/source/communication/tcp.cpp:89:31: error: ‘consteval void casual::common::communication::tcp::local::{anonymous}::socket::casual_enum_as_flag(Flag)’ declared ‘static’ but never defined [-Werror=unused-function]
   89 |                consteval void casual_enum_as_flag( Flag);
      |                               ^~~~~~~~~~~~~~~~~~~

if the function is not defined. If the function is defined it compiles withjout warning withoiut [[maybe_unused]]